internal/data/opentelemetry-proto-gen/common/v1/traceid_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewTraceID(t *testing.T) {
+	tid := NewTraceID(nil)
+	assert.EqualValues(t, []byte(nil), tid.id)
+	assert.EqualValues(t, 0, tid.Size())
+
+	b := []byte{1, 2, 3}
+	tid = NewTraceID(b)
+	assert.EqualValues(t, b, tid.id)
+	assert.EqualValues(t, b, tid.Bytes())
+	assert.EqualValues(t, 3, tid.Size())
+}
+
+func TestTraceIDHexString(t *testing.T) {
+	tid := NewTraceID(nil)
+	assert.EqualValues(t, "", tid.HexString())
+
+	tid = NewTraceID([]byte{0x12, 0x23, 0xAD})
+	assert.EqualValues(t, "1223ad", tid.HexString())
+}
+
+func TestTraceIDMarshal(t *testing.T) {
```

Comment:
I would usually split this test at least between binary vs json tests. Hard to read what is tested here. Also consider to split between error vs non-error.